### PR TITLE
Add minimum_index_compatibility_version, minimum_wire_compatibility_v…

### PR DESCRIPTION
**Pull request by mkleen**
…ersion to sys.node table

## Checklist

 - [X] User relevant changes are recorded in ``CHANGES.txt``
 - [x] Touched code is covered by tests
 - [x] Documentation has been updated if necessary
 - [X] [CLA](https://crate.io/community/contribute/cla/) is signed
 - [X] This does not contain breaking changes, or if it does:
    - It is released within a major release
    - It is recorded in ``CHANGES.txt``
    - It was marked as deprecated in an earlier release if possible
    - You've thought about the consequences and other components are adapted
      (E.g. AdminUI)

### CODE REVIEW

File: blackbox/docs/appendices/release-notes/unreleased.rst [hunk 1]
```diff
@@ -144,6 +144,10 @@ Deprecations
 Changes
 =======
 
+- Add `minimum_index_compatibility_version` and
+  `minimum_wire_compatibility_version` to sys.node table to expose current
```
Comment (mfussenegger) [inline blackbox/docs/appendices/release-notes/unreleased.rst]: please link to the relevant sections in the docs.. (which should be updated, see https://github.com/crate/crate/blob/master/blackbox/docs/admin/system-information.rst#nodes)

File: sql/src/main/java/io/crate/expression/reference/sys/node/NodeVersionStatsExpression.java [hunk 2]
```diff
@@ -47,5 +51,16 @@ public Boolean innerValue(NodeStatsContext nodeStatsContext) {
                 return nodeStatsContext.version().isSnapshot();
             }
         });
+        childImplementations.put(BUILD_SNAPSHOT, new SimpleNodeStatsExpression<Boolean>() {
```
Comment (mfussenegger) [inline sql/src/main/java/io/crate/expression/reference/sys/node/NodeVersionStatsExpression.java]: this is duplicate?

**Comment by mkleen** [comment]
> Shouldn't there be some test cases that require updating?

To my own surprise this is never tested anywhere, i though this is intentionally.

File: sql/src/main/java/io/crate/expression/reference/sys/node/NodeVersionStatsExpression.java [hunk 3]
```diff
@@ -47,5 +51,11 @@ public Boolean innerValue(NodeStatsContext nodeStatsContext) {
                 return nodeStatsContext.version().isSnapshot();
             }
         });
+        
+        childImplementations.put(MINIMUM_INDEX_COMPATIBILITY_VERSION,
+                                 constant(Version.CURRENT.minimumIndexCompatibilityVersion().toString()));
```
Comment (mfussenegger) [inline sql/src/main/java/io/crate/expression/reference/sys/node/NodeVersionStatsExpression.java]:
```suggestion
                                 constant(Version.CURRENT.minimumIndexCompatibilityVersion().externalNumber()));
```

Would make it explicit which display form we're using.

File: sql/src/main/java/io/crate/expression/reference/sys/node/NodeVersionStatsExpression.java [hunk 4]
```diff
@@ -47,5 +51,11 @@ public Boolean innerValue(NodeStatsContext nodeStatsContext) {
...
+
+        childImplementations.put(MINIMUM_WIRE_COMPATIBILITY_VERSION,
+                                 constant(Version.CURRENT.minimumCompatibilityVersion().toString()));
```
Comment (mfussenegger) [inline sql/src/main/java/io/crate/expression/reference/sys/node/NodeVersionStatsExpression.java]:
```suggestion
                                 constant(Version.CURRENT.minimumCompatibilityVersion().externalNumber()));
```

**Comment by mkleen** [comment]
retest this please

**Pull request merged**